delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/model/File.scala

```diff
@@ -37,32 +41,42 @@ final case class File(
     storageType: StorageType,
     attributes: FileAttributes,
     tags: Tags
-)
+) {
+  def metadata: Metadata = Metadata(tags.tags)
+}
 
 object File {
 
-  implicit def fileEncoder(implicit config: StorageTypeConfig): Encoder.AsObject[File] =
-    Encoder.encodeJsonObject.contramapObject { file =>
-      implicit val storageType: StorageType = file.storageType
-      val storageJson                       = Json.obj(
-        keywords.id  -> file.storage.iri.asJson,
-        keywords.tpe -> storageType.iri.asJson,
-        "_rev"       -> file.storage.rev.asJson
-      )
-      file.attributes.asJsonObject.add("_storage", storageJson)
-    }
+  final case class Metadata(tags: List[UserTag])
+
+  implicit def fileEncoder(implicit config: StorageTypeConfig): Encoder[File] = { file =>
```

Comment:
I noticed we're doing `Encoder.AsObject` all over the place - I'm not sure this is necessary? The `Json` ADT provides functions to work with objects (`mapObject` etc). It's designed so we operate on the abstraction - I think this adds unnecessary boilerplate?

Reply:
It does look nicer this way